### ISSUE TITLE
--sar switch has no effekt

### ISSUE THREAD

**Issue by stax76** [rigaya/NVEnc#8]
Hi,

I've noticed that the nvenc --sar switch has no effect. Maybe you can fix it or forward it to NVIDIA otherwise.

**Comment by rigaya** [comment]
I've fixed it in 3.08 / 3.09.
If it still occurs, please let me know the exact command line for NVEncC, as it may differ by the settings.

**Comment by stax76** [comment]
Thanks for the quick fix, I've tested avc and hevc output and confirm that it's fixed.

Did you change anything regarding error return code? I'm getting error code 1 even tough console output and output file look fine.

```
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
               Encoding using NVEncC 3.09 x64
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

@echo off
D:\Projekte\VS\VB\StaxRip\bin\Apps\NVEncC\NVEncC64.exe --cqp 20:23:25 --codec h265 --sar 2:1 -i "D:\Video\Samples\MKV\720p  SBR_temp\720p  SBR_new.avs" -o "D:\Video\Samples\MKV\720p  SBR_temp\720p  SBR_new_out.h265"

cmd.exe /C call "D:\Video\Samples\MKV\720p  SBR_temp\720p  SBR_new_NVEncC.bat"

NVEnc 3.09 (x64), using NVENC API v8.0
OS Version     Windows 10 x64 (15063)
CPU            Intel Core i7-6700K @ 4.00GHz [TB: 4.19GHz] (4C/8T)
GPU            #0: GeForce GTX 1060 6GB (10 EU) @ 1771 MHz (382.33)
Input Buffers  CUDA, 32 frames
Input Info     Avisynth 2.60(yv12)->nv12 [AVX2], 1280x720, 30/1 fps
Vpp Filters    copyHtoD
Output Info    H.265/HEVC main @ Level auto
1280x720p 2:1 30.000fps (30/1fps)
Encoder Preset default
Rate Control   CQP  I:20  P:23  B:25
Lookahead      off
GOP length     300 frames
B frames       0 frames
Ref frames     3 frames, LTR: off
AQ             off
CU max / min   auto / auto
Others         mv:auto
encoded 1232 frames, 680.66 fps, 4357.39 kbps, 21.33 MB
encode time 0:00:01, CPULoad: 41.19
frame type IDR    5
frame type I      5,  total size   0.32 MB
frame type P   1227,  total size  21.01 MB

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
            Error Encoding using NVEncC 3.09 x64
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

Encoding using NVEncC 3.09 x64 failed with exit code: 1 (0x1)

The exit code might be a system error code: STATUS_WAIT_1

The exit code might be a system error code: Unzulässige Funktion.

NVEnc 3.09 (x64), using NVENC API v8.0
OS Version     Windows 10 x64 (15063)
CPU            Intel Core i7-6700K @ 4.00GHz [TB: 4.19GHz] (4C/8T)
GPU            #0: GeForce GTX 1060 6GB (10 EU) @ 1771 MHz (382.33)
Input Buffers  CUDA, 32 frames
Input Info     Avisynth 2.60(yv12)->nv12 [AVX2], 1280x720, 30/1 fps
Vpp Filters    copyHtoD
Output Info    H.265/HEVC main @ Level auto
1280x720p 2:1 30.000fps (30/1fps)
Encoder Preset default
Rate Control   CQP  I:20  P:23  B:25
Lookahead      off
GOP length     300 frames
B frames       0 frames
Ref frames     3 frames, LTR: off
AQ             off
CU max / min   auto / auto
Others         mv:auto
encoded 1232 frames, 680.66 fps, 4357.39 kbps, 21.33 MB
encode time 0:00:01, CPULoad: 41.19
frame type IDR    5
frame type I      5,  total size   0.32 MB
frame type P   1227,  total size  21.01 MB
```

**Comment by rigaya** [comment]
I have reproduced the return code issue you have mentioned.
This is not intended, I'll check and fix.

**Comment by stax76** [comment]
It looks there is be a issue with --sar, I can only reproduce it with a particular file you can find here:

https://github.com/stax76/staxrip/issues/197

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                        Environment
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

StaxRip    : 1.5.0.3
Windows    : Windows 10 Pro 1703
Language   : German (Germany)
CPU        : Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
GPU        : NVIDIA GeForce GTX 1060 6GB, Intel(R) HD Graphics 530
Resolution : 3840 x 2160
DPI        : 288

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                   Source file MediaInfo
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

D:\Temp\StaxRip\Original file.mp4

General
Complete name             : D:\Temp\StaxRip\Original file.mp4
Format                    : MPEG-4
Format profile            : Base Media
Codec ID                  : isom (isom/iso2/avc1/mp41)
File size                 : 10.0 MiB
Duration                  : 5 s 419 ms
Overall bit rate mode     : Variable
Overall bit rate          : 15.5 Mb/s
Writing application       : Lavf54.63.104

Video
ID                        : 1
Format                    : AVC
Format/Info               : Advanced Video Codec
Format profile            : Main@L3.2
Format settings, CABAC    : Yes
Format settings, ReFrames : 3 frames
Format settings, GOP      : M=2, N=33
Codec ID                  : avc1
Codec ID/Info             : Advanced Video Coding
Duration                  : 5 s 400 ms
Bit rate mode             : Variable
Bit rate                  : 15.4 Mb/s
Maximum bit rate          : 18.0 Mb/s
**Width                     : 688 pixels
Height                    : 496 pixels
Display aspect ratio      : 1.387**
Frame rate mode           : Constant
Frame rate                : 25.000 FPS
Standard                  : PAL
Color space               : YUV
Chroma subsampling        : 4:2:0
Bit depth                 : 8 bits
Scan type                 : Progressive
Bits/(Pixel*Frame)        : 1.806
Stream size               : 9.92 MiB (99%)
Color range               : Limited

Audio
ID                        : 2
Format                    : AAC
Format/Info               : Advanced Audio Codec
Format profile            : LC
Codec ID                  : 40
Duration                  : 5 s 419 ms
Duration_LastFrame        : -21 ms
Bit rate mode             : Variable
Bit rate                  : 126 kb/s
Maximum bit rate          : 128 kb/s
Channel(s)                : 2 channels
Channel positions         : Front: L R
Sampling rate             : 48.0 kHz
Frame rate                : 46.875 FPS (1024 spf)
Compression mode          : Lossy
Stream size               : 83.2 KiB (1%)
Language                  : English
Default                   : Yes
Alternate group           : 1

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                           Script
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

LoadPlugin("D:\Projekte\VS\VB\StaxRip\bin\Apps\Plugins\avs\L-SMASH-Works\LSMASHSource.dll")
LSMASHVideoSource("D:\Temp\StaxRip\Original file.mp4", format = "YUV420P8")

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                     Script Properties
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

source frame count : 135
source frame rate  : 25.000000
source duration    : 00:00:05.4000000
target frame count : 135
target frame rate  : 25.000000
target duration    : 00:00:05.4000000

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
               Encoding using NVEncC 3.09 x64
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

@echo off
D:\Projekte\VS\VB\StaxRip\bin\Apps\NVEncC\NVEncC64.exe --cqp 20:23:25 **--sar 246:255** -i "D:\Temp\StaxRip\Original file_temp\Original file.avs" -o "D:\Temp\StaxRip\Original file_temp\Original file_out.h264"

cmd.exe /C call "D:\Temp\StaxRip\Original file_temp\Original file_NVEncC.bat"

NVEnc 3.09 (x64), using NVENC API v8.0
OS Version     Windows 10 x64 (15063)
CPU            Intel Core i7-6700K @ 4.00GHz [TB: 4.19GHz] (4C/8T)
GPU            #0: GeForce GTX 1060 6GB (10 EU) @ 1771 MHz (382.33)
Input Buffers  CUDA, 32 frames
Input Info     Avisynth 2.60(yv12)->nv12 [AVX2], 688x496, 25/1 fps
Vpp Filters    copyHtoD
Output Info    H.264/AVC high @ Level auto
688x496p 1776850:2148757 25.000fps (25/1fps)
Encoder Preset default
Rate Control   CQP  I:20  P:23  B:25
Lookahead      off
GOP length     250 frames
B frames       3 frames
Ref frames     3 frames, LTR: off
AQ             off
Others         mv:auto cabac deblock adapt-transform:auto bdirect:auto
encoded 135 frames, 537.85 fps, 1333.71 kbps, 0.86 MB
encode time 0:00:00, CPULoad: 59.31
frame type IDR   1
frame type I     1,  total size  0.02 MB
frame type P    34,  total size  0.36 MB
frame type B   100,  total size  0.47 MB

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
            Error Encoding using NVEncC 3.09 x64
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

Encoding using NVEncC 3.09 x64 failed with exit code: 1 (0x1)

The exit code might be a system error code: STATUS_WAIT_1

The exit code might be a system error code: Unzulässige Funktion.

NVEnc 3.09 (x64), using NVENC API v8.0
OS Version     Windows 10 x64 (15063)
CPU            Intel Core i7-6700K @ 4.00GHz [TB: 4.19GHz] (4C/8T)
GPU            #0: GeForce GTX 1060 6GB (10 EU) @ 1771 MHz (382.33)
Input Buffers  CUDA, 32 frames
Input Info     Avisynth 2.60(yv12)->nv12 [AVX2], 688x496, 25/1 fps
Vpp Filters    copyHtoD
Output Info    H.264/AVC high @ Level auto
688x496p 1776850:2148757 25.000fps (25/1fps)
Encoder Preset default
Rate Control   CQP  I:20  P:23  B:25
Lookahead      off
GOP length     250 frames
B frames       3 frames
Ref frames     3 frames, LTR: off
AQ             off
Others         mv:auto cabac deblock adapt-transform:auto bdirect:auto
encoded 135 frames, 537.85 fps, 1333.71 kbps, 0.86 MB
encode time 0:00:00, CPULoad: 59.31
frame type IDR   1
frame type I     1,  total size  0.02 MB
frame type P    34,  total size  0.36 MB
frame type B   100,  total size  0.47 MB

Start:    00:35:32
End:      00:35:32
Duration: 00:00:00

General

Complete name             : D:\Temp\StaxRip\Original file_temp\Original file_out.h264
Format                    : AVC
Format/Info               : Advanced Video Codec
File size                 : 879 KiB
Duration                  : 5 s 400 ms
Overall bit rate          : 1 334 kb/s

Video

Format                    : AVC
Format/Info               : Advanced Video Codec
Format profile            : High@L3
Format settings, CABAC    : Yes
Format settings, ReFrames : 3 frames
Duration                  : 5 s 400 ms
Bit rate                  : 1 334 kb/s
Width                     : 688 pixels
Height                    : 496 pixels
**Display aspect ratio      : 0.198**
Frame rate                : 25.000 FPS
Color space               : YUV
Chroma subsampling        : 4:2:0
Bit depth                 : 8 bits
Scan type                 : Progressive
Bits/(Pixel*Frame)        : 0.156
Stream size               : 879 KiB (100%)

**Comment by rigaya** [comment]
Now return code issue is fixed with 3.10.

About sar in the movie above, --sar 246:255 is not correct for 688x496 movie to be shown in 4:3, and resulted in the internal "display area" calculation to overflow. If you use the correct --sar 124:129, you can have the 4:3 display.

If calculating sar is troublesome, you can use --dar 4:3, which makes NVEncC to automatically set the correct sar.

**Comment by stax76** [comment]
Thanks for the new build.

**Comment by stax76** [comment]
I don't understand why my par value wasn't correct, the user wanted a dar with 1.34 and my par value gave exactly that. I've changed it to use --dar instead but this don't work either.

https://github.com/stax76/staxrip/blob/master/Encoding/NVIDIAEncoder.vb#L419

```

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
      _________ __                __________.__        
     /   _____//  |______  ___  __\______   \__|_____  
     \_____  \\   __\__  \ \  \/  /|       _/  \____ \ 
     /        \|  |  / __ \_>    < |    |   \  |  |_> >
    /_______  /|__| (____  /__/\_ \|____|_  /__|   __/ 
            \/           \/      \/       \/   |__|   

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                        Environment
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

StaxRip    : 1.5.0.5
Windows    : Windows 10 Pro 1703
Language   : German (Germany)
CPU        : Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
GPU        : NVIDIA GeForce GTX 1060 6GB, Intel(R) HD Graphics 530
Resolution : 3840 x 2160
DPI        : 288

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                   Source file MediaInfo
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

D:\Temp\StaxRip\SD  anamorphic.mpg

General
Complete name                      : D:\Temp\StaxRip\SD  anamorphic.mpg
Format                             : MPEG-PS
File size                          : 15.0 MiB
Duration                           : 39 s 552 ms
Overall bit rate mode              : Variable
Overall bit rate                   : 3 187 kb/s

Video
ID                                 : 224 (0xE0)
Format                             : MPEG Video
Format version                     : Version 2
Format profile                     : Main@Main
Format settings, BVOP              : Yes
Format settings, Matrix            : Custom
Format settings, GOP               : M=3, N=12
Format settings, picture structure : Frame
Duration                           : 39 s 400 ms
Bit rate mode                      : Variable
Bit rate                           : 2 932 kb/s
Width                              : 720 pixels
Height                             : 576 pixels
Display aspect ratio               : 16:9
Frame rate                         : 25.000 FPS
Standard                           : PAL
Color space                        : YUV
Chroma subsampling                 : 4:2:0
Bit depth                          : 8 bits
Scan type                          : Interlaced
Scan order                         : Top Field First
Compression mode                   : Lossy
Bits/(Pixel*Frame)                 : 0.283
Time code of first frame           : 05:27:31:02
Time code source                   : Group of pictures header
GOP, Open/Closed                   : Open
Stream size                        : 13.8 MiB (92%)

Audio
ID                                 : 192 (0xC0)
Format                             : MPEG Audio
Format version                     : Version 1
Format profile                     : Layer 2
Duration                           : 39 s 552 ms
Bit rate mode                      : Constant
Bit rate                           : 192 kb/s
Channel(s)                         : 2 channels
Sampling rate                      : 48.0 kHz
Compression mode                   : Lossy
Delay relative to video            : -641 ms
Stream size                        : 927 KiB (6%)

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                          ProjectX
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

C:\ProgramData\Oracle\Java\javapath\Java.exe -jar "D:\Projekte\VS\VB\StaxRip\bin\Apps\ProjectX\ProjectX.jar" "D:\Temp\StaxRip\SD  anamorphic.mpg" -out "D:\Temp\StaxRip\SD  anamorphic_temp\\"

Version: ProjectX 0.91.0.00/31.03.2011
Reading GUI-Switch...
Reading Help Switch...
Reading Config File Switch...
Start without GUI...
Loading last Config or Standard File...
Fehler beim Laden der ini Datei: java.io.FileNotFoundException: D:\Projekte\VS\VB\StaxRip\bin\X.ini (Das System kann die angegebene Datei nicht finden)
Loading Language -> 'en'
ProjectX 0.91.0.00/31.03.2011 TEST PROJECT ONLY , User: Frank
TERMS AND CONDITIONS:
(1) this is a free Java based demux utility.
(2) It is intended for educational purposes only, as a non-commercial test project.
(3) released under the terms of the GNU GPL.
(4) there is NO WARRANTY of any kind attached to this software.
(5) use it at your own risk and for your own education.
Java Environment
13. Juni 2017    01:13:31 MESZ
java.version	1.8.0_131
java.vendor	Oracle Corporation
java.home	C:\Program Files (x86)\Java\jre1.8.0_131
java.vm.version	25.131-b11
java.vm.vendor	Oracle Corporation
java.vm.name	Java HotSpot(TM) Client VM
java.class.vers	52.0
java.class.path	D:\Projekte\VS\VB\StaxRip\bin\Apps\ProjectX\ProjectX.jar
os.name	Windows 10
os.arch	x86
os.version	10.0
ini.file	D:\Projekte\VS\VB\StaxRip\bin\X.ini
ext.disk.access	disabled or library not found
user.language	en
user.name	Frank
user.home	C:\Users\Frank
quick CL usage:
Note: CL doesn't load the GUI components, except with switch [-gui]
<without options>  ...starts the GUI
switches and inputfiles can be in any order
options:
[-ini <path + inifile>] ..use that specified iniFile instead of the standard
[-dvx1] ..create a .d2v ProjectFile on demux
[-dvx2] ..create a .d2v ProjectFile + .ac3.wav (RIFF WAVE Header)
[-dvx3] ..create a .d2v ProjectFile + .mpa.wav (RIFF WAVE Header)
[-dvx4] ..create a .d2v ProjectFile + .ac3.wav + mpa.wav (RIFF WAVE Header)
[-out <path>] ..use that specified directory for output
[-name <filename>] ..use that specified filename for output
[-cut <file>] ..use that text based file as cutpoint list
[-chp <file>] ..use that text based file as chapterpoint list
[-id <tokens>] ..use only these (P)IDs, separated by comma ","
[-gui] ..display the GUI using all given CLI options
[-log] ..write the normal logfile
[-saveini] ..save changes made bei CLI in active .ini
[-split <xxx>] ..split output at xxx MB
[-demux, -tom2p, -topva, -tovdr, -tots, -filter] ..action types
Loading Basic Classes...
Multicolor OFF / switches 00000000
Reading CLI Switches...
Preparing GUI (if enabled)...
Check Commons-Net library access...
Loading AC3 frames...
Starting Collection Process...
preparing collection(s)...
1 %<<< session infos >>>
Dienstag, 13. Juni 2017  01:13 Uhr MESZ
ProjectX 0.91.0.00 (31.03.2011)
-> working with collection 0
-> save normal log file
-> log 'packets out of sequence' / bit errors
-> log 'missing startcodes'
-> log 'PES header found in ES'
-> log WSS
-> log VPS
-> log RDS
-> log max. 500 warnings/errors
-> write all video data
-> write all other data
-> patch c.d.flagged infos of pictures
-> add sequence end code
-> set resolution in SDE
-> PVA: strictly specs. for audio streams
-> VOB: determine diff. Cell timelines
-> TS: ignore scrambled packets
-> TS: enhanced search for open packets
-> TS: join file segments (of Dreambox®)
-> TS: generate PMT stream dependent
-> get only enclosed PES/TS packets
-> concatenate different recordings
-> ensure 1st PES-packet start with video
-> generate PCR/SCR from PTS
-> write output files to: 'D:\Temp\StaxRip\SD  anamorphic_temp'
-> main I/O-Buffersize in bytes 4096000 bytes
FileSegments:
* (0) D:\Temp\StaxRip\SD  anamorphic.mpg
second. Files:
* ---
+> Input File 0:  'D:\Temp\StaxRip\SD  anamorphic.mpg' (15.758.223 bytes)
-> Filetype is MPEG-2 PS/SS (PES Container)
-> demux
demuxing A/V PES file SD  anamorphic.mpg
-> found PES-ID 0xE0 (MPEG Video) @ 32
-> found PES-ID 0xC0 (MPEG Audio) @ 40173
-> video basics: 720*576 @ 25fps @ 0.7031 (16:9) @ 3202800 bps - vbv 112
-> starting export of video data @ GOP# 0
!> dropping useless B-Frames @ GOP# 0 / new Timecode 00:00:00.000
100 %-> skip system_end_code @ 15758219
++> Mpg Video: PID 0x0000 / PesID 0xE0 / SubID 0x00 :
-> Video: fr-ct-1p-cg-og-dg -> 982-1-0-82-0-0
-> Video length: 982 frames @ 00:00:39.280
-> GOP summary: min. 12, max. 36 fields; contains interlaced frames
-> avg. nom. bitrate 2998707bps (min/max: 2121600/5264800)
-> set first sequenceheader bitrate to 5264800bps
---> new File: D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.m2v
++> Mpg Audio: PID 0x0000 / PesID 0xC0 / SubID 0x00 :
check & synchronize audio file  SD  anamorphic.mp
1 %-> check CRC of AC-3 / MPEG-Audio L1,2
-> remove CRC in MPEG-Audio L1,2
-> add frames
-> Audio PTS: first packet 00:00:01.358, last packet 00:00:40.718
-> Video PTS: start 1.GOP 00:00:02.000, end last GOP 00:00:41.280
-> adjusting audio at video-timeline
2 %-> src_audio: MPEG-1, Layer2, 48000Hz, stereo, 192kbps, CRC @ 00:00:00.000
100 %!> 16 frame(s) (384ms) added @ 00:00:38.904
audio frames: wri-pre-skip-ins-add 1637-0-0-0-16 @ 00:00:39.288 done...
---> new File: 'D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.mp2'
summary of created media files:
.Video (m2v):	982 Frames	00:00:39.280		'D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.m2v'
Audio 00 (mp2):	1637 Frames	00:00:39.288	0-0-0-16	'D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.mp2'
=> 15.666.564 bytes written...
-> we have 2 warnings/errors.
done...    1 collection(s) processed  @ 00:00:00.384

Start:    01:13:31
End:      01:13:32
Duration: 00:00:00

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                    Index with ffmsindex
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

D:\Projekte\VS\VB\StaxRip\bin\Apps\Plugins\both\ffms2\ffmsindex.exe "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.m2v" "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.ffindex"

Writing index... done.

Start:    01:13:32
End:      01:13:32
Duration: 00:00:00

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                           Script
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

LoadPlugin("D:\Projekte\VS\VB\StaxRip\bin\Apps\Plugins\both\ffms2\ffms2.dll")
FFVideoSource("D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.m2v", colorspace = "YV12", \
              cachefile = "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.ffindex")

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                     Script Properties
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

source frame count : 982
source frame rate  : 25.000000
source duration    : 00:00:39.2800000
target frame count : 982
target frame rate  : 25.000000
target duration    : 00:00:39.2800000

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
               Encoding using NVEncC 3.10 x64
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

@echo off
D:\Projekte\VS\VB\StaxRip\bin\Apps\NVEncC\NVEncC64.exe --cqp 20:23:25 --dar 911681:500000 -i "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic.avs" -o "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_out.h264"

cmd.exe /C call "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_NVEncC.bat"

NVEnc 3.10 (x64), using NVENC API v8.0
OS Version     Windows 10 x64 (15063)
CPU            Intel Core i7-6700K @ 4.00GHz (4C/8T)
GPU            #0: GeForce GTX 1060 6GB (10 EU) @ 1771 MHz (382.33)
Input Buffers  CUDA, 32 frames
Input Info     Avisynth 2.60(yv12)->nv12 [AVX2], 720x576, 25/1 fps
Vpp Filters    copyHtoD
Output Info    H.264/AVC high @ Level auto
720x576p 102257:23784 25.000fps (25/1fps)
Encoder Preset default
Rate Control   CQP  I:20  P:23  B:25
Lookahead      off
GOP length     250 frames
B frames       3 frames
Ref frames     3 frames, LTR: off
AQ             off
Others         mv:auto cabac deblock adapt-transform:auto bdirect:auto
encoded 982 frames, 1412.95 fps, 770.00 kbps, 3.61 MB
encode time 0:00:00, CPULoad: 35.69
frame type IDR   4
frame type I     4,  total size  0.08 MB
frame type P   247,  total size  1.68 MB
frame type B   731,  total size  1.84 MB

Start:    01:19:27
End:      01:19:28
Duration: 00:00:01

General
Complete name             : D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_out.h264
Format                    : AVC
Format/Info               : Advanced Video Codec
File size                 : 3.61 MiB

Video
Format                    : AVC
Format/Info               : Advanced Video Codec
Format profile            : High@L3
Format settings, CABAC    : Yes
Format settings, ReFrames : 3 frames
Width                     : 720 pixels
Height                    : 576 pixels
Display aspect ratio      : 1.930
Frame rate                : 25.000 FPS
Standard                  : PAL
Color space               : YUV
Chroma subsampling        : 4:2:0
Bit depth                 : 8 bits
Scan type                 : Progressive

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
              Muxing using mkvmerge 12.0.0 x64
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

D:\Projekte\VS\VB\StaxRip\bin\Apps\MKVToolNix\mkvmerge.exe -o "D:\Temp\StaxRip\SD  anamorphic.mkv" "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_out.h264" --global-tags "D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_tags.xml" --ui-language en

mkvmerge v12.0.0 ('Trust / Lust') 64bit
'D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_out.h264': Using the demultiplexer for the format 'AVC/h.264'.
'D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_out.h264' track 0: Using the output module for the format 'AVC/h.264 (unframed)'.
The file 'D:\Temp\StaxRip\SD  anamorphic.mkv' has been opened for writing.
'D:\Temp\StaxRip\SD  anamorphic_temp\SD  anamorphic_out.h264' track 0: Extracted the aspect ratio information from the MPEG-4 layer 10 (AVC) video data and set the display dimensions to 1112/576.
The cue entries (the index) are being written...
Multiplexing took 0 seconds.

Start:    01:19:28
End:      01:19:29
Duration: 00:00:00

General
Complete name                 : D:\Temp\StaxRip\SD  anamorphic.mkv
Format                        : Matroska
Format version                : Version 4 / Version 2
File size                     : 3.62 MiB
Duration                      : 39 s 280 ms
Overall bit rate              : 773 kb/s
Encoded date                  : UTC 2017-06-12 23:19:29
Writing application           : mkvmerge v12.0.0 ('Trust / Lust') 64bit
Writing library               : libebml v1.3.4 + libmatroska v1.4.5
Writing frontend              : StaxRip 1.5.0.5

Video
ID                            : 1
Format                        : AVC
Format/Info                   : Advanced Video Codec
Format profile                : High@L3
Format settings, CABAC        : Yes
Format settings, ReFrames     : 3 frames
Codec ID                      : V_MPEG4/ISO/AVC
Duration                      : 39 s 280 ms
Bit rate                      : 770 kb/s
Width                         : 720 pixels
Height                        : 576 pixels
Display aspect ratio          : 1.931
Original display aspect ratio : 1.930
Frame rate mode               : Constant
Frame rate                    : 25.000 FPS
Standard                      : PAL
Color space                   : YUV
Chroma subsampling            : 4:2:0
Bit depth                     : 8 bits
Scan type                     : Progressive
Bits/(Pixel*Frame)            : 0.074
Stream size                   : 3.61 MiB (100%)
Default                       : Yes
Forced                        : No

-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
                        Job Complete
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

Start:    01:19:27
End:      01:19:29
Duration: 00:00:01
```

**Comment by rigaya** [comment]
I've fixed this issue on 3.11 with eeddb2c,

720x576 with --dar 911681:500000
688x496 with --sar 246:255

can now be accepted, and using dar is not necessary.

**Comment by stax76** [comment]
Thanks for the new version and support, 3-4 test I made were all successful.
